lib/frame.rb

```diff
@@ -0,0 +1,44 @@
+class Frame 
+
+attr_reader :knocked_pins, :frame, :frame_id, :frame_score, :strike, :spare
+
+  def initialize
+    @frame_id = 0
+    @frame = {}
+  end
+
+  def first_roll(knocked_pins1)
+    @knocked_pins = knocked_pins1
+    if knocked_pins == 10
+       @frame = {frame_id: @frame_id += 1, first_roll: 'strike' }
+    else
+      @frame = {frame_id: @frame_id += 1, first_roll: knocked_pins1 }
+    end
+    over
+  end
+
+  def second_roll(knocked_pins2)
+    @knocked_pins += knocked_pins2
+    if @knocked_pins == 10
+      @frame[:second_roll] = 'spare'
+    else
+      @frame[:second_roll] = knocked_pins2
+    over
+    end
+  end
+  
+  def score
+    
+
+  end
+  
+
+  def over
+    if @knocked_pins > 10 
+       raise 'Wrong number - re-type!'
+    end
+  end
+  
+end
+
+
```

Comment:
code is well named, makes sense what you were trying to achieve here. 

Couple of thoughts on this: 
- you could simplify some of the conditionals to be one liners
- your error catcher for if a wrong number is entered could be at the top of the first and second roll methods so that no logic is executed before an error is caught.